addons/enemybar/subtargetBar.lua

```diff
@@ -0,0 +1,80 @@
+--[[
+Copyright © 2015, Mike McKee
+All rights reserved.
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of enemybar nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Mike McKee BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+--]]
+
+stbg_cap_l = images.new(stbg_cap_settings)
+stbg_cap_r = images.new(stbg_cap_settings)
+stbg_body = images.new(stbg_body_settings)
+stfg_body = images.new(stfg_body_settings)
+st_text = texts.new(' ${name|(Name)}', settings.textSettings, settings)
+
+init_subtarget_images = function(...)
+    stbg_cap_l:pos(settings.pos.x + 399, settings.pos.y + 15)
+    stbg_cap_r:pos(settings.pos.x + settings.subtargetBarWidth + 400, settings.pos.y + 15)
+    stbg_body:pos(settings.pos.x + 400, settings.pos.y + 15)
+    stfg_body:pos(settings.pos.x + 400, settings.pos.y + 15)
+    st_text:pos(settings.pos.x + 400, settings.pos.y + 15)
+end
+
+render_subtarget_bar = function(...)
+    if visible == true then
+        local target = windower.ffxi.get_mob_by_target('t')
+        local subtarget = windower.ffxi.get_mob_by_target('st')
+        local player = windower.ffxi.get_player()
```

Comment:
I fixed the get_player issue. I moved it inside the if, and cached the id which only really benefits sub-target since I don't use the animation which relies on player.in_combat)

I think as for the second part, I'll just worry about this later.